### PR TITLE
Add ECVRF cli

### PR DESCRIPTION
**Pull request by jonas-lj**
Closing [#8069](https://github.com/MystenLabs/sui/issues/8069) in sui

### CODE REVIEW

File: cryptotool/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "cryptotool"
```
Comment (benr-ml) [inline cryptotool/Cargo.toml:2]: perhaps let's put this under `fastcrypto-cli`, and name the tool `ecvrf-cli`?

File: fastcrypto-cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "fastcrypto-cli"
+version = "0.1.0"
+license = "Apache-2.0"
+authors = ["Mysten Labs <build@mystenlabs.com>"]
+edition = "2021"
+publish = ["crates-io"]
+description = "zk-proof library used at Mysten Labs"
```
Comment (benr-ml) [inline fastcrypto-cli/Cargo.toml:8]: pls update

File: fastcrypto-cli/src/ecvrf.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    /// Hex-encoded Sha512 hash of the proof. Must be 64 bytes.
+    #[clap(short, long)]
+    output: String,
+
+    /// Encoding of the proof to verify. Must be 80 bytes.
+    #[clap(short, long)]
+    proof: String,
+
+    /// Hex encoding of the input string used to generate the proof. May be of arbitrary length but
+    /// should ideally be at least 16 bytes.
```
Comment (benr-ml) [inline fastcrypto-cli/src/ecvrf.rs:46]: why is that preferred?

File: fastcrypto-cli/src/ecvrf.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    Error::new(ErrorKind::Other, "Failed to serialize secret key.")
+                })?);
+            let pk_string =
+                hex::encode(bincode::serialize(&keypair.pk).map_err(|_| {
+                    Error::new(ErrorKind::Other, "Failed to serialize public key.")
+                })?);
+
+            let mut result = "Secret key: ".to_string();
+            result.push_str(&sk_string);
+            result.push_str("\nPublic key: ");
```
Comment (benr-ml) [inline fastcrypto-cli/src/ecvrf.rs:83]: Verification Key (like the name used below)

File: fastcrypto-cli/src/ecvrf.rs (hunk 3 above)
Comment (jonas-lj) [reply fastcrypto-cli/src/ecvrf.rs:46]: If the prover can predict the input or it has very low entropy, the prover can create a keypair giving a specific output. @kchalkias had a tweet about it last week. But this warning should either be omitted be more clearly written.

File: fastcrypto-cli/src/ecvrf.rs (hunk 4 above)
Comment (jonas-lj) [reply fastcrypto-cli/src/ecvrf.rs:82]: I've used public key everywhere instead to align with the fastcrypto api.

File: fastcrypto-cli/src/ecvrf.rs (hunk 3 above)
Comment (jonas-lj) [reply fastcrypto-cli/src/ecvrf.rs]: I've removed it. This warning is use case specific and should be somewhere else.

**Review by benr-ml — APPROVED** [review]

**Pull request merged**